### DISCUSSION TITLE
Indexing into timeline variables - Goal: Show different timelines across conditions

### DISCUSSION THREAD

**Discussion by notesmith** [jspsych/jsPsych#2341] [Q&A]
I am trying to do something where each stimulus is paired with a condition (and the condition determines which order the blocks are presented in). My current set up has all of the stimuli are in the timeline variable so when I call it, all of the stimuli are presented for condition 1 and then condition 2, etc. I only want one part of the timevariable to called in each one which I think is why I need to index into the timeline variable, but I cannot figure out how to do that. Many thanks in advance! 

How I set up my timeline variable: 
After a series of pushing and flattening lists; I combine all of them into timeline Var: 
```
var timeline_vars = [];

all_img.forEach(function(el, ind) {
  timeline_vars.push({stimulus: el.stimulus, ML1: all_ML1[ind].ML1, .... condition: list_cond[ind].condition}); // many other stimuli in here! 
})

```

This is how I am currently setting up my data which works, but repeats all of the stimuli (since I am calling in the entire timeline variable) when a condition is met. I am looping across conditions: 

```
var arrayLength = all_img.length
  
  procedure_list = []
  for(let i = 0; i < arrayLength; i++){
      var condition_here= condition_assignment[i]; //THIS ASSUMES THIS CORRESPONDS WHICH IT SHOULD
      //alert(condition_here)

      if (condition_here === 1){
        var procedure_temp = {
          timeline: [block1,block2],
          timeline_variables: timeline_vars}}
      if (condition_here === 2){
        var procedure_temp = {
          timeline: [block2,block1],
          timeline_variables: timeline_vars}}
      if (condition_here === 3){
        var procedure_temp = {
          timeline: [block2,block1],
          timeline_variables: timeline_vars}}
      procedure_list.push(procedure_temp)}
or (var i = 0; i < arrayLength; i++) {
  timeline_temp.push(procedure_list[i])}
 
  jsPsych.init({
    //timeline: [native_speaker,procedure_list[1]], //UPDATE
    // timeline: function() {
    //   for (var i = 0; i < arrayLength; i++) {
    //     procedure_list[i]}},
    timeline: timeline_temp,
    timeline_variables: timeline_vars,
    //experiment_structure: procedure_list,
    on_finish: function(){jsPsych.data.displayData();}
  });
```

These are the other attempts that I thought would fix my problem, but that I cannot get to work: 

```
  for(let i = 0; i < arrayLength; i++){
      var condition_here= condition_assignment[i];
      //alert(condition_here)
      var procedure_temp = {
        timeline_variables: timeline_vars,
        timeline: function(){
          if (condition_here === 1){
            return [block1,block2]
        }
          if (condition_here === 2){
            return [block2,block1]
        }
        if (condition_here === 3){
            return [block2,block1]
        }
      }}   
      procedure_list.push(procedure_temp)} /// This one doesn't load

  for(let i = 0; i < arrayLength; i++){
      var condition_here= condition_assignment[i];

      if (condition_here === 1){
        timeline_temp.push(block1,block2)}
      if (condition_here === 2){
        timeline_temp.push(block1,block2)}} /// This one doesn't work because no timeline variable is included
```
  
It seems like a plausible solutin would be to take the first solution I showed and then somehow index by i into timelinevariables? Like timelineVar[i] etc. This doesn't seem to work.
Please let me know if you would need more information for how I set up timelineVar to help! It's a huge chunk of code so I figured this could be enough.

Thanks!!

**Comment by csavasegal** [comment]
Not sure why the other two do not work, but the way to index could be to do: 
`timeline_variables: timeline_vars.slice(i,i+1)}}`
